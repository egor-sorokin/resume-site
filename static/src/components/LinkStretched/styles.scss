@import '../../styles/base/_vars.scss';

.link-stretched {
  letter-spacing: 3px;

  transform: translateX(-40px);
  transition: transform $defaultTiming $defaultEasing,
  letter-spacing $defaultTiming $defaultEasing;
}

.link-stretched:hover {
  letter-spacing: 5px;

  transform: translateX(-34px);
}
