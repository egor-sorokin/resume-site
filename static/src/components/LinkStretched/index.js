import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = props => {
  const {text, url, target, cssClasses, onClick} = props;

  const handleClick = (e) => {
    if (!url) {
      e.preventDefault();
      onClick();
    } else {
      onClick(e.currentTarget.dataset.anchorId.substring(1), true);
    }
  };

  return (
    <a
      href={url}
      data-anchor-id={url}
      target={target}
      onClick={handleClick}
      className={'link-stretched ' + cssClasses}
    >{text}</a>
  )
};


LinkStretched.propTypes = {
  onClick: PropTypes.func,
  cssClasses: PropTypes.string,
  text: PropTypes.string,
  url: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  cssClasses: '',
  text: '',
  url: '',
  target: '',
  onClick: () => {
  }
};


export default LinkStretched;