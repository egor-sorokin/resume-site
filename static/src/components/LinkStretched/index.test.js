// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import LinkStretched from './index';

describe('<LinkStretched />', () => {
  let component, onClick;
  const text = 'Text test';
  const url = 'test.url';
  const event = {
    preventDefault: () => {
    },
    currentTarget: {
      dataset: {
        anchorId: 'test.url'
      }
    }
  };


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<LinkStretched />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });
    
    
    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 0 components', () => {
      expect(component.children()).toHaveLength(0);
    });
  });


  describe('traversions tests', () => {
    beforeEach(() => {
      component = shallow(<LinkStretched text={text}/>);
    });


    it('checks a link text', () => {
      expect(component.find('.link-stretched').text()).toEqual(text);
    });
  });


  describe('link functionality tests', () => {
    beforeEach(() => {
      jest.spyOn(event, 'preventDefault');
      onClick = jest.fn(() => 'mock implementation');
    });


    it('clicks on an anchor link', () => {
      component = shallow(<LinkStretched text={text} url={url} onClick={onClick}/>);

      const linkButton = component.find('a').first();
      linkButton.simulate('click', event);

      expect(onClick).toHaveBeenCalled();
      expect(onClick).toHaveBeenCalledTimes(1);
    });


    it('clicks on a button link', () => {
      component = shallow(<LinkStretched text={text} onClick={onClick}/>);

      const linkButton = component.find('a').first();
      linkButton.simulate('click', event);

      expect(event.preventDefault).toBeCalled();
      expect(onClick).toHaveBeenCalled();
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
