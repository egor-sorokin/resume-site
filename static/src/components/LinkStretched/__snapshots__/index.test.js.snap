// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkStretched /> base tests matches snapshot 1`] = `
<a
  className=" link-stretched"
  data-anchor-id=""
  href=""
  onClick={[Function]}
  target=""
/>
`;
